server/storage/endpoint/gc_key_space.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2022 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package endpoint
+
+import (
+	"encoding/json"
+	"math"
+	"strings"
+	"time"
+
+	"github.com/pingcap/errors"
+	"go.etcd.io/etcd/clientv3"
+)
+
+// KeySpaceGCSafePoint is gcWorker's safepoint for specific key-space
+type KeySpaceGCSafePoint struct {
+	SpaceID   string `json:"space_id"`
+	SafePoint uint64 `json:"safe_point"`
+}
+
+// KeySpaceGCSafePointStorage defines the storage operations on KeySpaces' safe points
+type KeySpaceGCSafePointStorage interface {
+	// Service safe point interfaces.
+	SaveServiceSafePointByKeySpace(spaceID string, ssp *ServiceSafePoint) error
```

Comment:
```suggestion
	SaveServiceSafePoint(spaceID string, ssp *ServiceSafePoint) error
```

Reply:
It might be confused with `SaveServiceGCSafePoint` from `GCSafePointStorage` as @pingyu suggest: https://github.com/tikv/pd/pull/4937#discussion_r871321826
Maybe we could use a simpler suffix like `KS`? How about `SaveServiceSafePointKS`